.github/workflows/reusable-release.yml

```diff
@@ -0,0 +1,34 @@
+name: Reusable Release Workflow
+
+on:
+  workflow_call:
+    secrets:
+      NODE_AUTH_TOKEN:
+        required: true
+
+env:
+  CI: true
+jobs:
+  publish-npm:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Prepare for publication to npm
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          registry-url: "https://registry.npmjs.org"
+          cache: npm
+      - run: npm ci
+      - if: ${{ contains(github.ref_name, 'beta') }}
+        run: npm publish --access public --tag beta
```

Comment:
This assumes that there is a pre-publish build step, which is a fine assumption to make (it currently holds for all of our repositories), but I'd like it to be written down here.